pymobiledevice3/services/webinspector.py

```
@@ -179,6 +190,19 @@ def get_open_pages(self):
                 apps[self.connected_application[app].name] = self.application_pages[app].values()
         return apps
 
+    def wait_for_application(self, timeout: Union[int, float] = 5):
+        start_time = time.time()
+        while not self.connected_application:
+            self.flush_input()
+            if time.time() - start_time >= timeout:
+                raise TimeoutError(f'Unable to find connected application after {timeout} second(s)')
+
+    def wait_for_applications_and_reload_pages(self, timeout: Union[int, float] = 5):
```

Timeout of what? I don't think this function is needed

---

if the app does not opened, we want to raise an exception and not stuck.

so we want to give a timeout for the operation.

example of this is if the device is locked with passcode...

anyway, I changed it now that the function _wait_for_application will raise the exception :)